operator/src/main/java/oracle/kubernetes/operator/steps/ShutdownManagedServerStep.java

```diff
@@ -0,0 +1,249 @@
+   * @param pod server pod
+   * @return asynchronous step
+   */
+  public static Step createShutdownManagedServerStep(Step next, String serverName, V1Pod pod) {
+    return new ShutdownManagedServerStep(next, serverName, pod);
+  }
+
+  private static String getManagedServerShutdownPath(Boolean isGracefulShutdown) {
+    StringBuilder stringBuilder = new StringBuilder("/management/weblogic/latest/serverRuntime/");
+    String shutdownString = isGracefulShutdown ? "shutdown" : "forceShutdown";
+    return stringBuilder.append(shutdownString).toString();
+  }
+
+  private static String getManagedServerShutdownPayload(Boolean isGracefulShutdown,
+      Boolean ignoreSessions, Long timeout, Boolean waitForAllSessions) {
+    if (!isGracefulShutdown) {
+      return "{}";
+    }
+
+    StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder.append("{  \"ignoreSessions\": ")
+        .append(ignoreSessions)
+        .append(", \"timeout\": ")
+        .append(timeout)
+        .append(", \"waitForAllSessions\": ")
+        .append(waitForAllSessions)
+        .append("}");
+
+    return stringBuilder.toString();
+  }
+
+  @Override
+  public NextAction apply(Packet packet) {
+    DomainPresenceInfo info = packet.getSpi(DomainPresenceInfo.class);
+    V1Service service = info.getServerService(serverName);
+
+    if (service == null) {
+      return doNext(packet);
+    } else {
+      return doNext(
+            Step.chain(
+                SecretHelper.createAuthorizationSourceStep(),
+                new ShutdownManagedServerWithHttpStep(service, pod, getNext())),
+            packet);
+    }
+  }
+
+  static final class ShutdownManagedServerProcessing extends HttpRequestProcessing {
+
+    ShutdownManagedServerProcessing(Packet packet, @Nonnull V1Service service, V1Pod pod) {
+      super(packet, service, pod);
+    }
+
+    private HttpRequest createRequest() {
+      String serverName = pod.getMetadata().getLabels().get(LabelConstants.SERVERNAME_LABEL);
+      String clusterName = getClusterNameFromServiceLabel();
+
+      DomainPresenceInfo info = packet.getSpi(DomainPresenceInfo.class);
+      Shutdown shutdown = Optional.ofNullable(info.getDomain().getServer(serverName, clusterName))
+          .map(ServerSpec::getShutdown).orElse(null);
+
+      Boolean isGracefulShutdown = isGracefulShutdown(shutdown);
+      Long timeout = getTimeout(shutdown);
+      Boolean ignoreSessions = getIgnoreSessions(shutdown);
+      Boolean waitForAllSessions = getWaitForAllSessions(shutdown);
+
+      HttpRequest request = createRequestBuilder(getRequestUrl(isGracefulShutdown))
+            .POST(HttpRequest.BodyPublishers.ofString(getManagedServerShutdownPayload(
+                isGracefulShutdown, ignoreSessions, timeout, waitForAllSessions))).build();
+      return request;
+    }
+
+    private Boolean isGracefulShutdown(Shutdown shutdown) {
+      String shutdownType = Optional.ofNullable(shutdown).map(Shutdown::getShutdownType)
```

Comment:
I could be wrong, but this doesn't appear to check if the env var is set for the pod, or if graceful shutdown is configured on WebLogic server mbean itself...

Reply:
Yes, you're right.  The configuration in the step is just looking at the domain resource shutdown properties and  not checking the environment.  Let me see how I can get to the env vars.

Reply:
Changed so that we take the environment variable as precedence.  By the time we are deleting a server pod, the environment variable would already be populated with the configuration from the shutdown configuration of the domain resource but the code will still check the shutdown configuration if the environment variable doesn't exist.